### DISCUSSION TITLE
Configuring FCST_PCP_COMBINE_INPUT_ACCUMS correctly

### DISCUSSION THREAD

**Discussion by MarcelCaron-NOAA** [dtcenter/METplus#1801] [Configuration] [answered]
I'm trying to set up a PCPCombine configuration file with some flexibility as to the input forecast precipitation accumulation field that METplus uses, but METplus gives me an error.  Am I using the FCST_PCP_COMBINE_INPUT_ACCUMS correctly?  

Here is the error (full error in the attached log file):
`ERROR  : get_field() -> can't get data plane from file "/lfs/h1/ops/prod/com/hiresw/v8.1/hiresw.20220910/hiresw.t18z.arw_5km.f14.ak.grib2"`

As I currently have PCPCombine configured, METplus will use the first listed accumulation (ex. A24) and then throw an error when the accumulation isn't available.  In this case, I would have preferred METplus attempted to use the second (A3) and then third listed accumulation (A1) using the third accumulation only, since the latter was the only available precip accumulation field in that file.

I'm attaching a log file that includes the full error message from a recent job on NCEP's WCOSS2 computer, the config file I used on that run, and the environment variables referenced in the configuration file.

[LOG_PCPCombine_fcsthireswarw_precip_mrms_job82_for2022091108_runon20220912212916.log](https://github.com/dtcenter/METplus/files/9558154/LOG_PCPCombine_fcsthireswarw_precip_mrms_job82_for2022091108_runon20220912212916.log)
[CONFIG_PCPCombine_fcsthireswarw_precip_mrms_job82_for2022091108_runon20220912212916.txt](https://github.com/dtcenter/METplus/files/9558156/CONFIG_PCPCombine_fcsthireswarw_precip_mrms_job82_for2022091108_runon20220912212916.txt)
[ENV_VARS_job82.txt](https://github.com/dtcenter/METplus/files/9558346/ENV_VARS_job82.txt)

**Comment by georgemccabe** [comment, marked as answer]
Hi @MarcelCaron-NOAA, the PCPCombine wrapper does support providing a list of accumulations in order of preference. However, if the accumulation (level) is not referenced in the filename template, the logic does not know which accumulation is available and uses the first one in the list. The wrapper logic currently does not support looking into the actual file contents to determine what is available to use. So it looks like you are using the functionality correctly, but the filenames do not allow this to work as expected.

The MET tool pcp_combine had a -sum mode that determines which files/data to use, however I believe that you can only provide a single input accumulation level for this mode. @JohnHalleyGotway, is that correct? If so, do you have any suggestions as to how this could be accomplished? If not, how would you configure the tool to allow multiple accumulation intervals based on what is available in the files?

**Reply by JohnHalleyGotway** [reply]
@MarcelCaron-NOAA and @georgemccabe, it is true that pcp_combine's "sum" option requires the user to specify a desired input accumulation.

For example, if you want a 48 hour accumulation computed from 24-hour input accumulations, it'll search for 2 matching files. Or 4 matching files for a 12-hour input accumulation. Or 8 files for a 6-hour input accumulation, and so on.

If you want to MIX input accumulations, i.e. 24-hour accum = 12-hour + 6-hour + 3-hour + 3-hour, the only want to do this is directly on the command line using the "-add" option... like this:
```
pcp_combine -add \
file1 'name="APCP"; level="A12";' \
file2 'name="APCP"; level="A6";' \
file3 'name="APCP"; level="A3";' \
file4 'name="APCP"; level="A3";' \
APCP_24A.nc
```

So you can explicitly specify what data should be read from each input file. But as George mentions, METplus would need to actually know what each file contains to be able to configure a call to pcp_combine like this. And METplus does not "inventory" the data in each input file to figure all this out.

I don't know what the best way forward is here.

**Reply by MarcelCaron-NOAA** [reply]
@georgemccabe and @JohnHalleyGotway:  I see, I should be reading in files that include the level name in them if I’d like to provide MET with a list of accumulations.  Given the input file name templates I'm using don't include the level name, I think the way forward for me will be to explicitly define the accumulation intervals to use for a run rather than having METplus identify them for me.  Thanks for your responses.
